Application/Commands/CreateWorkEntryCommand.cs

```diff
@@ -21,25 +24,36 @@ public class CreateWorkEntryCommand
 {
     private readonly TenantAppDbContext _context;
     private readonly IClaimsProvider _claimsProvider;
+    private readonly IAssignmentsApi _assignmentsApi;
 
     public CreateWorkEntryCommand(
         TenantAppDbContext context,
-        IClaimsProvider claimsProvider
+        IClaimsProvider claimsProvider,
+        IAssignmentsApi assignmentsApi
     )
     {
         _context = context;
         _claimsProvider = claimsProvider;
+        _assignmentsApi = assignmentsApi;
     }
 
     public async Task<long> ExecuteAsync(CreateWorkEntryCommandParams createWorkEntryCommandParams)
     {
+        var project = _assignmentsApi.FindEmployeeProjectAsync(createWorkEntryCommandParams.ProjectId);
+
+        if (project == null)
+        {
+            throw new ArgumentException("This project doesn't exist");
```

Comment:
Do not use ArgumentException here, create a dedicated exception type (for example, `ProjectNotAvailableException`)